config/environments/production.rb

```diff
@@ -85,3 +85,6 @@
   # Do not dump schema after migrations.
   config.active_record.dump_schema_after_migration = false
 end
+
+# Required when using Rails.application.routes.url_helpers from outside the request/response life cycle (models, jobs, lib)
+Rails.application.routes.default_url_options = { host: 'era.library.ualberta.ca', port: 443, protocol: 'https' }
```

Comment:
The production.rb in ansible is using variables, not ENV VARs for replacement.  A minor point of clarification for sure. I agree that a good direction would be to set the values in the environments configs to be environment variables the way you described for the secrets and then have the ansible playbook set their values. Makes sense to me!